src/components/progress-report/workflow/progress-report-workflow.edgedb.repository.ts

```diff
@@ -0,0 +1,193 @@
+import { Injectable } from '@nestjs/common';
+import {
+  ID,
+  NotFoundException,
+  PublicOf,
+  RichTextDocument,
+  Session,
+  UnsecuredDto,
+} from '~/common';
+import { e, RepoFor } from '~/core/edgedb';
+import { ProgressReportStatus as Status } from '../dto';
+import { ProgressReportWorkflowEvent } from './dto/workflow-event.dto';
+import { ProgressReportWorkflowRepository } from './progress-report-workflow.repository';
+import { InternalTransition } from './transitions';
+
+@Injectable()
+export class ProgressReportWorkflowEdgeDBRepository
+  extends RepoFor(ProgressReportWorkflowEvent, {
+    hydrate: (event) => ({
+      ...event['*'],
+      report: true,
+      who: true,
+      transition: event.transitionId,
+    }),
+  }).customize((cls, { defaults }) => {
+    return class extends cls {
+      static omit = [defaults.create, defaults.update, defaults.delete];
+      async list(
+        reportId: ID,
+        _session: Session,
+      ): Promise<Array<UnsecuredDto<ProgressReportWorkflowEvent>>> {
+        const query = e.select(e.ProgressReport, () => ({
+          filter_single: { id: reportId },
+          workflowEvents: {
+            id: true,
+            at: true,
+            status: true,
+            transitionId: true,
+            notes: true,
+            who: {
+              id: true,
+              email: true,
+            },
+          },
+        }));
+        const results = await this.db.run(query);
+        if (!results) {
+          throw new NotFoundException('Could not find report', 'reportId');
+        }
+        if (!results.workflowEvents) {
+          throw new NotFoundException(
+            'Could not find workflow events for report',
+            'reportId',
+          );
+        }
+        return results.workflowEvents.map((event) => ({
+          id: event.id,
+          at: event.at,
+          status: event.status,
+          transition: event.transitionId,
+          notes: event.notes,
+          who: {
+            id: event.who.id,
+            email: event.who.email,
+          },
+        }));
+      }
+    };
+  })
+  implements PublicOf<ProgressReportWorkflowRepository>
+{
+  async recordTransition(
```

Comment:
@CarsonF - Would it make sense to move both `recordTransition` and `recordBypass` up to the service so that they don't need to be duplicated here?  Neither of them are actually doing anything database specific.

Reply:
Sure